File: Facebook Chatbot (Python)/ngramnode.py
# The node will contain a list of words following
# 1 given word (bigram) or 2 given words (trigram)
# and the counts of the frequencies of each of the following words
class NgramNode:
    def __init__(self):
        self.wordsAfter = list()
        self.wordsAfterCounts = list()

    def addToWordsAfter(self, word):
        #If word already in wordsAfter, increments its count & return
        for i in range(0, len(self.wordsAfter)):
            if self.wordsAfter[i] == word:
                self.wordsAfterCounts[i] = self.wordsAfterCounts[i] + 1
                return

        #If new word, add to words after & initialize its count to 1
        self.wordsAfter.append(word)
        self.wordsAfterCounts.append(1)

    def getWordsAfter(self):
        return self.wordsAfter

    def getWordsAfterCounts(self):
        return self.wordsAfterCounts

File: Chatbot (Java)/Chatbot/NgramNode.java
package Chatbot;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

/* The node will contain a list of words following
 * 1 given word (bigram) or 2 given words (trigram)
 * and the counts of the frequencies of each of the following words
 */
public class NgramNode implements Serializable {
	private List<String> wordsAfter;
	private List<Integer> wordsAfterCounts;
	
	public NgramNode() {
		wordsAfter = new ArrayList<String>();
		wordsAfterCounts = new ArrayList<Integer>();
	}
	
	public void addToWordsAfter(String word) {
		//If word already in wordsAfter, increments its count & return
		for(int i=0; i<wordsAfter.size(); i++) {
			if (wordsAfter.get(i).equals(word)) {
				wordsAfterCounts.set(i, wordsAfterCounts.get(i) + 1);
				return;
			}
		}
		
		//If new word, add to words after & initialize its count to 1
		wordsAfter.add(word);
		wordsAfterCounts.add(1);
	}
	
	public List<String> getWordsAfter() {
		return wordsAfter;
	}
	
	public List<Integer> getWordsAfterCounts() {
		return wordsAfterCounts;
	}
}